src/common/transformations/include/transformations/common_optimizations/enable_shapeof_constant_folding.hpp

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/pass/graph_rewrite.hpp>
+#include <transformations_visibility.hpp>
+
+namespace ov {
+namespace pass {
+
```

Comment:
Could you please add some comments here about transformation. What does it actually do? Maybe why we need it? Any use cases? Just in a couple of words.